packages/cli/src/databases/migrations/common/1675940580449-PurgeInvalidWorkflowConnections.ts

```diff
@@ -1,60 +1,11 @@
-import type { WorkflowEntity } from '@/databases/entities/workflow-entity';
-import type { MigrationContext, IrreversibleMigration } from '@/databases/types';
+import { UserError } from 'n8n-workflow';
 
-interface Workflow {
-	id: number;
-	nodes: WorkflowEntity['nodes'] | string;
-	connections: WorkflowEntity['connections'] | string;
-}
+import type { IrreversibleMigration } from '@/databases/types';
 
 export class PurgeInvalidWorkflowConnections1675940580449 implements IrreversibleMigration {
-	async up({ escape, parseJson, runQuery, nodeTypes }: MigrationContext) {
-		const workflowsTable = escape.tableName('workflow_entity');
-		const workflows: Workflow[] = await runQuery(
-			`SELECT id, nodes, connections FROM ${workflowsTable}`,
-		);
-
-		await Promise.all(
-			workflows.map(async (workflow) => {
-				const connections = parseJson(workflow.connections);
-				const nodes = parseJson(workflow.nodes);
-
-				const nodesThatCannotReceiveInput = nodes.reduce<string[]>((acc, node) => {
-					try {
-						const nodeType = nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
-						if ((nodeType.description.inputs?.length ?? []) === 0) {
-							acc.push(node.name);
-						}
-					} catch (error) {}
-					return acc;
-				}, []);
-
-				Object.keys(connections).forEach((sourceNodeName) => {
-					const connection = connections[sourceNodeName];
-					const outputs = Object.keys(connection);
-
-					outputs.forEach((outputConnectionName /* Like `main` */) => {
-						const outputConnection = connection[outputConnectionName];
-
-						// It filters out all connections that are connected to a node that cannot receive input
-						outputConnection.forEach((outputConnectionItem, outputConnectionItemIdx) => {
-							outputConnection[outputConnectionItemIdx] = (outputConnectionItem ?? []).filter(
-								(outgoingConnections) =>
-									!nodesThatCannotReceiveInput.includes(outgoingConnections.node),
-							);
-						});
-					});
-				});
-
-				// Update database with new connections
-				return await runQuery(
-					`UPDATE ${workflowsTable} SET connections = :connections WHERE id = :id`,
-					{
-						connections: JSON.stringify(connections),
-						id: workflow.id,
-					},
-				);
-			}),
+	async up() {
+		throw new UserError(
+			'Migration "PurgeInvalidWorkflowConnections1675940580449" is no longer supported. Please upgrade to n8n@1.0.0 first.',
```

Comment:
```suggestion
		on a second thought, this migration will still run for every new user, and on tests.
Maybe we should update the migration to something like:
 ```suggestion
		const workflowCount = await queryRunner.manager.count(WorkflowEntity);
		if (workflowCount) {
			throw new UserError(
				'Migration "PurgeInvalidWorkflowConnections1675940580449" is no longer supported. Please upgrade to n8n@1.0.0 first.',
			);
		}
```
```